### PR TITLE
Added get multiple api for badger

### PR DESCRIPTION
**Pull request by ghost**
Added get multiple api in badger.
Able to get upto 60% performance improvement in ldbc s10 dataset queries.
4.69 seconds -> 2.61 seconds

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -756,6 +756,42 @@ func (db *DB) getMemTables() ([]*memTable, func()) {
...
+			continue
+		}
+		version := y.ParseTs(key)
+		for i := 0; i < len(tables); i++ {
+			vs := tables[i].sl.Get(key)
+			y.NumMemtableGetsAdd(db.opt.MetricsEnabled, 1)
+			if vs.Meta == 0 && vs.Value == nil {
+				continue
+			}
+			// Found the required version of the key, return immediately.
```
Comment (kevinmingtarja) [inline db.go:780]: This comment should be refactored, since we don't return immediately unlike in `db.get()`

File: discard_test.go [hunk 2]
```diff
@@ -17,12 +17,171 @@
...
+}
+
+// bytesLessThan compares two byte arrays lexicographically.
+func bytesLessThan(a, b []byte) bool {
+	return bytes.Compare(a, b) >= 0
+}
+
+func TestReadC(t *testing.T) {
+	t.Skip()
+	allKeysF, err := os.Open("/home/harshil/all_keys_2")
```
Comment (tauraamui) [inline discard_test.go:77]: Bit weird to have an explicit user path, make it a temp path or something?

Comment (harshil-goel) [reply discard_test.go:76]: This s just a temp test for my local validation. Hence the t.Skip() before the test starts. I can't commit this as the directory being used is quite big.

File: level_handler.go [hunk 3]
```diff
@@ -273,6 +273,113 @@ func (s *levelHandler) getTableForKey(key []byte) ([]*table.Table, func() error)
...
+			if done[i] {
+				continue
+			}
+			results[i], err = s.get(key)
+			if err != nil {
+				return results, err
+			}
+		}
+		return results, nil
+	} else {
```
Comment (sanjayk-github-dev) [inline level_handler.go:344]: @harshil-goel I need to understand this code block ... will sync up with you offline.

File: level_handler.go [hunk 4]
```diff
@@ -273,6 +273,113 @@ func (s *levelHandler) getTableForKey(key []byte) ([]*table.Table, func() error)
 	return []*table.Table{tbl}, tbl.DecrRef
 }
 
+func (s *levelHandler) getBatch(keys [][]byte, done []bool) ([]y.ValueStruct, error) {
+	// Find the table for which the key is in, and then seek it
+	getForKey := func(key []byte) (y.ValueStruct, func() error, []*table.Iterator) {
+		tables, decr := s.getTableForKey(key)
+		keyNoTs := y.ParseKey(key)
```
Comment (mangalaman93) [inline level_handler.go:280]: we should just parse the key once, we are doing it again and again

File: level_handler.go [hunk 5]
```diff
@@ -273,6 +273,113 @@ func (s *levelHandler) getTableForKey(key []byte) ([]*table.Table, func() error)
...
+	}
+
+	// Use old results from getForKey and find in those tables.
+	findInIter := func(key []byte, itrs []*table.Iterator) y.ValueStruct {
+		var maxVs y.ValueStruct
+
+		for _, it := range itrs {
+			it.Seek(key)
+			if !it.Valid() {
+				continue
```
Comment (mangalaman93) [inline level_handler.go:317]: break?

**Review by mangalaman93 — CHANGES_REQUESTED** [review]
This PR needs lot more refactoring, testing and cleanup. Only two comments for now but once it is cleaned up, can review again.

**Pull request closed without merging**